From khali@linux-fr.org Sat Nov 26 19:23:57 2005
Date: Sat, 26 Nov 2005 20:18:43 +0100
From: Jean Delvare <khali@linux-fr.org>
To: Greg KH <greg@kroah.com>
Cc: Peter Korsgaard <jacmet@sunsite.dk>
Subject: [PATCH 05/25] i2c: Add support for Barco LTP->DVI to i2c-parport
Message-Id: <20051126201843.75fd91a3.khali@linux-fr.org>
Content-Disposition: inline; filename=i2c-parport-barco-ltp-dvi.patch

From: Peter Korsgaard <jacmet@sunsite.dk>

The following patch adds support for the Barco LTP->DVI I2C adapter to
the i2c-parport driver.

Signed-off-by: Peter Korsgaard <jacmet@sunsite.dk>
Signed-off-by: Jean Delvare <khali@linux-fr.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 Documentation/i2c/busses/i2c-parport |    1 +
 drivers/i2c/busses/i2c-parport.h     |   12 +++++++++++-
 2 files changed, 12 insertions(+), 1 deletion(-)

--- gregkh-2.6.orig/Documentation/i2c/busses/i2c-parport
+++ gregkh-2.6/Documentation/i2c/busses/i2c-parport
@@ -17,6 +17,7 @@ It currently supports the following devi
  * Velleman K8000 adapter
  * ELV adapter
  * Analog Devices evaluation boards (ADM1025, ADM1030, ADM1031, ADM1032)
+ * Barco LPT->DVI (K5800236) adapter
 
 These devices use different pinout configurations, so you have to tell
 the driver what you have, using the type module parameter. There is no
--- gregkh-2.6.orig/drivers/i2c/busses/i2c-parport.h
+++ gregkh-2.6/drivers/i2c/busses/i2c-parport.h
@@ -80,6 +80,14 @@ static struct adapter_parm adapter_parm[
 		.setscl	= { 0x01, DATA, 1 },
 		.getsda	= { 0x10, STAT, 1 },
 	},
+	/* type 6: Barco LPT->DVI (K5800236) adapter */
+	{
+		.setsda	= { 0x02, DATA, 1 },
+		.setscl	= { 0x01, DATA, 1 },
+		.getsda	= { 0x20, STAT, 0 },
+		.getscl	= { 0x40, STAT, 0 },
+		.init	= { 0xfc, DATA, 0 },
+	},
 };
 
 static int type;
@@ -91,4 +99,6 @@ MODULE_PARM_DESC(type,
 	" 2 = Velleman K8000 adapter\n"
 	" 3 = ELV adapter\n"
 	" 4 = ADM1032 evaluation board\n"
-	" 5 = ADM1025, ADM1030 and ADM1031 evaluation boards\n");
+	" 5 = ADM1025, ADM1030 and ADM1031 evaluation boards\n"
+	" 6 = Barco LPT->DVI (K5800236) adapter\n"
+);
